lib/ansible/plugins/connection/ssh.py

```diff
@@ -801,30 +801,33 @@ def _file_transport_command(self, in_path, out_path, sftp_action):
         success = False
         for method in methods:
             returncode = stdout = stderr = None
-            if method == 'sftp':
-                cmd = self._build_command('sftp', to_bytes(host))
-                in_data = u"{0} {1} {2}\n".format(sftp_action, shlex_quote(in_path), shlex_quote(out_path))
-                in_data = to_bytes(in_data, nonstring='passthru')
-                (returncode, stdout, stderr) = self._run(cmd, in_data, checkrc=False)
-            elif method == 'scp':
-                if sftp_action == 'get':
-                    cmd = self._build_command('scp', u'{0}:{1}'.format(host, shlex_quote(in_path)), out_path)
-                else:
-                    cmd = self._build_command('scp', in_path, u'{0}:{1}'.format(host, shlex_quote(out_path)))
-                in_data = None
-                (returncode, stdout, stderr) = self._run(cmd, in_data, checkrc=False)
-            elif method == 'piped':
-                if sftp_action == 'get':
-                    # we pass sudoable=False to disable pty allocation, which
-                    # would end up mixing stdout/stderr and screwing with newlines
-                    (returncode, stdout, stderr) = self.exec_command('dd if=%s bs=%s' % (in_path, BUFSIZE), sudoable=False)
-                    out_file = open(to_bytes(out_path, errors='surrogate_or_strict'), 'wb+')
-                    out_file.write(stdout)
-                    out_file.close()
-                else:
-                    in_data = open(to_bytes(in_path, errors='surrogate_or_strict'), 'rb').read()
+            try:
```

Comment:
We try and wrap the smallest amount of code possible in a try/except.  Our preference would be to only wrap the `_run` and `exec_command`  lines in try/except.

Once this is done, we'll need to review again, but overall this approach looks correct.